### ISSUE TITLE
Don't use a default kube-context

### ISSUE THREAD

**Issue by shadiramadan** [GoogleContainerTools/skaffold#4140]
From the documentation:
> By default, Skaffold uses the current kube-context from your kube-config file.

IMO this is very dangerous and is an accident waiting to happen.

All it would take is forgetting you switched to a production context to cause some very annoying problems for someone.

I think it should always be explicitly set by default.

**Comment by dgageot** [comment]
Thank you @shadiramadan. I understand your concern, however, skaffold behaves like kubectl or helm or basically any other tool in that domain. I don't think we want the users to have to always specify a kube-context

**Comment by pre** [comment]
There is also a related bug: if you have a namespace defined in your current context of kube-config, skaffold sync will fail for assets deployed not in this exact namespace. See https://github.com/GoogleContainerTools/skaffold/issues/1668#issuecomment-595752550 

These practically mean that (at for me) skaffold needs to be wrapped around a script like this: https://github.com/GoogleContainerTools/skaffold/issues/1668#issuecomment-596386312

Wrapping skaffold provides also a safety net for not accidentally running it against production.

**Comment by stanislav-zaprudskiy** [comment]
Would setting to deploy only in a valid context work? You can achieve this via profiles activation - https://skaffold.dev/docs/environment/profiles/#activation. We just set up all deployments inside the profiles section, which gets activated only when in a proper context (`docker-desktop`, `microk8s`, etc) is used by kube-config.

**Comment by tstromberg** [comment]
Thanks for opening this!

I also agree with @dgageot on this one. Never point your default to production, otherwise you will invariably mutate production accidentally. This goes not only for skaffold, but for any tool that interacts with a cluster. 

Someone will invariably shoot themselves in the foot with the defaults some day, but forcing a specific context would make skaffold more confusing for folks who don't know the context name of their local dev environment.

**Comment by shadiramadan** [comment]
No problem @tstromberg, @dgageot.  I made this ticket out of me having a quick panic when I was about to run a skaffold command and thought to myself- how does it know which context to use?

I ended up explicitly setting the `kubeContext` field that @stanislav-zaprudskiy  pointed out to `minikube`.

I'm trialing skaffold as I want to drop docker-compose and have our team develop on k8s directly.
It has taken a bit of time as I have had to create custom kustomize patches for our dev environment to mount NFS volumes instead of local-storage or EFS drives...  But I'm almost there.

I'm probably a ways away from trusting skaffold for deployments but developing on it is the first step :).
